app/command_giphy.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/platform/model"
+	goi18n "github.com/nicksnyder/go-i18n/i18n"
+)
+
+type GiphyProvider struct {
+}
+
+const (
+	CMD_GIPHY = "giphy"
+)
+
+func init() {
+	RegisterCommandProvider(&GiphyProvider{})
+}
+
+func (me *GiphyProvider) GetTrigger() string {
+	return CMD_GIPHY
+}
+
+func (me *GiphyProvider) GetCommand(T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_GIPHY,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_giphy.desc"),
+		AutoCompleteHint: T("api.command_giphy.hint"),
+		DisplayName:      T("api.command_giphy.name"),
+	}
+}
+
+type giphyTranslateResponse struct {
```

Comment:
This response object is kind of huge. I only actually use one or two fields. Didn't bother to see if I could put only a handful of these fields in this struct. Pro tip: [this website](https://mholt.github.io/json-to-go/) converts an arbitrary JSON object into a Golang struct.

Reply:
Yeah, you can (and probably should) exclude everything you don't care about.